Sensus.Shared.iOS/Callbacks/UILocalNotifications/UILocalNotificationCallbackScheduler.cs

```diff
@@ -13,77 +13,85 @@
+using System.Collections.Generic;
+
 using UIKit;
 using Xamarin.Forms.Platform.iOS;
 
 namespace Sensus.Shared.iOS.Callbacks.UILocalNotifications
 {
     public class UILocalNotificationCallbackScheduler : iOSCallbackScheduler
     {
-        private Dictionary<string, UILocalNotification> _callbackIdNotification;
+        private readonly Dictionary<string, UILocalNotification> _callbackIdNotification;
 
         public UILocalNotificationCallbackScheduler()
         {
             _callbackIdNotification = new Dictionary<string, UILocalNotification>();
         }
 
-        protected override void ScheduleCallbackAsync(string callbackId, int delayMS, bool repeating, int repeatDelayMS, bool repeatLag)
-        {
-            // the following lines need to precede the run on main thread to avoid deadlocks -- this is an old comment. not sure if it's still the case.
-            string userNotificationMessage = GetCallbackUserNotificationMessage(callbackId);
-            string notificationId = GetCallbackNotificationId(callbackId);
+        protected override void ScheduleCallbackAsync(ICallbackData meta)
+        {
+            if (meta.CallbackId == null || SensusContext.Current.ActivationId == null) return;
 
             SensusContext.Current.MainThreadSynchronizer.ExecuteThreadSafe(() =>
-            {
-                NSDictionary callbackInfo = GetCallbackInfo(callbackId, repeating, repeatDelayMS, repeatLag, notificationId, SensusContext.Current.ActivationId);
-
-                // user info can be null (e.g., if we don't have an activation ID). don't schedule the notification if this happens.
-                if (callbackInfo == null)
-                    return;
-
-                // all properties below were introduced in iOS 8.0. we currently target 8.0 and above, so these should be safe to set.
-                UILocalNotification callbackNotification = new UILocalNotification
+            {
+                // all properties below were introduced in iOS 8.0. we currently target 8.0 and above, so these should be safe to set.
+                var callbackNotification = new UILocalNotification
                 {
-                    FireDate = DateTime.UtcNow.AddMilliseconds(delayMS).ToNSDate(),
+                    FireDate = DateTime.UtcNow.Add(meta.RepeatDelay).ToNSDate(),
                     TimeZone = null,  // null for UTC interpretation of FireDate
-                    AlertBody = userNotificationMessage,
-                    UserInfo = callbackInfo
+                    AlertBody = GetCallbackUserNotificationMessage(meta.CallbackId)
+                };
+
+                new iOSCallbackData(callbackNotification)
+                {
+                    CallbackId = meta.CallbackId,
+                    IsRepeating = meta.IsRepeating,
+                    RepeatDelay = meta.RepeatDelay,
+                    LagAllowed = meta.LagAllowed,
+                    ActivationId = SensusContext.Current.ActivationId
                 };
```

Comment:
That's actually the exact code I had in mind when I mentioned not being completely happy with this class while we were meeting today. I was just playing around with a new pattern to see how it felt. In the end I think I've decided it is more confusing than helpful.